File: bubble/bubble.erl
-module(bubble).
-export([sort/1]).

sort([]) -> [];
sort(List) ->
  [L | Tail] = lists:reverse(switch(List)),
  [L | sort(Tail)].

switch([]) -> [];
switch([X]) -> [X];
switch([X,Y | Tail]) when X > Y ->
  [Y | switch([X|Tail])];
switch([X,Y | Tail])  -> 
  [X | switch([Y|Tail])].

File: bubble/bubble.exs
defmodule Bubble do
  @doc """
  ## buble sort
  """
  def sort([]), do: [] 

  def sort(list) do 
    [h | tail] = Enum.reverse(switch(list))
    [h | sort(tail)]
  end


  def switch([a,b |tail]) when a > b do
    [b | switch([a | tail])]
  end
  def switch([a,b |tail]) do
    [a | switch([b | tail])]
  end
  def switch([a]) do 
    [a]
  end
end
